src/persist/src/indexed/runtime.rs

```diff
@@ -360,7 +372,9 @@ mod tests {
             let mut persister = registry.open("path", "restart-1")?;
             let (_, meta) = persister.create_or_load("0")?.into_inner();
             let mut snap = meta.snapshot()?;
-            assert_eq!(snap.read_to_end(), data);
+            let mut actual = snap.read_to_end();
```

Comment:
theres a couple other calls to `read_to_end` in this file not sure if they all need to call `sort`

Reply:
Put the sort back in read_to_end using an extionsion trait to Snapshot